File: examples/C/testit.c
#include <czmq.h>
#define CHUNK_SIZE  250000

int main (void)
{
    FILE *file = fopen ("testdata", "r");
    assert (file);

    size_t offset = 0;
    byte *data = malloc (CHUNK_SIZE);
    assert (data);
    while (true) {
        fseek (file, offset, SEEK_SET);
        size_t size = fread (data, 1, CHUNK_SIZE, file);
        if (size == 0)
            break;
        offset += CHUNK_SIZE;
    }
    free (data);
    fclose (file);
    return 0;
}

File: examples/Python/testit.py
import os

CHUNK_SIZE = 250000

def main():
    file = open("testdata", "rb")

    offset = 0

    while True:
        file.seek(offset, os.SEEK_SET)
        data = file.read(CHUNK_SIZE)
        if not data:
            break
        offset += len(data)

    file.close()

if __name__ == '__main__':
    main()
